### DISCUSSION TITLE
Can't resolve fs

### DISCUSSION THREAD

**Discussion by roljohntorralba** [vercel/next.js#16567] [Help] [answered]
Hello, newbie here.

I've tried creating a sample app to learn Next, I tried following the tutorial and imported `fs` on one of my pages, but I ran on some errors, specifically:
```
error - ./lib/posts.js:1:0
Module not found: Can't resolve 'fs'
> 1 | import fs from 'fs'
```

I installed Next JS using `npx create-next-app` tried to add some pages, and tried importing `fs`.

This is the file that causes it:
```
import fs from 'fs'
import path from 'path'

export function getPostsData() {
    return {
        title: "Hello"
    }
}
```

I have already finished the [Learn](https://nextjs.org/learn/basics/create-nextjs-app?utm_source=next-site&utm_medium=homepage-cta&utm_campaign=next-website) tutorial, and everything worked fine, even though it uses `fs`.

So, yeah, the tutorial files (next 9.3.5) worked with `fs` while the new app I created (next 9.5.2) doesn't.

I tried debugging the scripts and got this:
```
Error: > Build failed because of webpack errors
    at build (/Users/roljohntorralba/Sites/lumad/node_modules/next/dist/build/index.js:15:918)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {stack: 'Error: > Build failed because of webpack erro…ctions (internal/process/task_queues.js:93:5)', message: '> Build failed because of webpack errors'}
```

Did I miss anything? What should I do?

**Comment by rafaelalmeidatk** [comment, marked as answer]
`fs` is a Node.js dependency that isn't available in the browser. Next.js will automatically eliminate server dependencies as long as you use them in the data-fetching methods ([see this](https://next-code-elimination.now.sh/)). In your snippet you are importing `fs` but aren't using it inside the function, and probably you aren't using this function in `getStaticProps` either, so Next.js can't eliminate the import, giving you the error

**Reply by MoOx** [reply]
See https://github.com/vercel/next.js/issues/16153#issuecomment-673374488: in short Node import & code MUST be in `pages` folder, otherwise not correctly removed for the frontend
